dev/NavigationView/NavigationView.cpp

```diff
@@ -2434,44 +2434,14 @@ winrt::IInspectable NavigationView::MenuItemFromContainer(winrt::DependencyObjec
 {
     if (container)
     {
-        if (IsTopNavigationView())
+        if (auto const nvib = container.try_as<winrt::NavigationViewItemBase>())
         {
-            // Search topnav first, if not found, search overflow
-            if (auto ir = m_topNavRepeater.get())
-            {
-                if (auto element = container.try_as<winrt::UIElement>())
-                {
-                    auto index = ir.GetElementIndex(element);
-                    if (index != -1)
-                    {
-                        return GetItemFromIndex(ir, index);
-                    }
-                }
-            }
-
-            if (auto ir = m_topNavRepeaterOverflowView.get())
+            if (auto const parentRepeater = GetParentItemsRepeaterForContainer(nvib))
```

Comment:
why weren't we using this helper before? Was it just an oversight?

Reply:
yep, it was an oversight

---
In reply to: [401132750](https://github.com/microsoft/microsoft-ui-xaml/pull/2196#discussion_r401132750) [](ancestors = 401132750)